rockcraft.yaml

```diff
@@ -0,0 +1,163 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+---
+
+name: charmed-ranger-rock
+base: ubuntu@22.04
+version: 2.4.0-22.04-edge
+summary: Charmed ranger ROCK OCI
+description: |
+  Apache Ranger™ is a framework to enable, monitor and
+  manage comprehensive data security across the Hadoop platform.
+license: Apache-2.0
+
+platforms:
+  amd64:
+
+parts:
+  base:
+    plugin: nil
+    source: "https://downloads.apache.org/ranger/2.4.0/apache-ranger-2.4.0.tar.gz" # yamllint disable-line
+    build-packages:
+      - build-essential
+      - curl
+      - gcc
+      - gnupg
+      - patch
+      - wget
+      - maven
+      - git
+      - openjdk-11-jdk-headless
+    build-environment:
+      - SOURCE: "https://downloads.apache.org/ranger/"
+      - VERSION: "2.4.0"
+      - TAR_FILE: "apache-ranger-2.4.0.tar.gz"
+      - JAVA_HOME: /usr/lib/jvm/java-11-openjdk-amd64
+      - RANGER_DIST: /home/ranger/dist
+    override-build: |
+      craftctl default
+
+      # make home directory
+      mkdir -p ${RANGER_DIST}
+
+      # Download the Ranger tar
+      wget "${SOURCE}/${VERSION}/${TAR_FILE}" -O ranger.tar.gz
+      wget "${SOURCE}/${VERSION}/${TAR_FILE}.sha512" -O ranger.tar.sha512
```

Comment:
not really important, but it might be better to store the sha512 here, instead of downloading it

Reply:
in theory, if the archive has been altered, the 'hacker' can alter the sha512 too. Storing the sha locally once provides a slightly better proof that the archive is good. I am saying slightly better cause we could already copy the corrupted sha, but at least it is done once, so there is less probability than pulling it each time.

Reply:
Ah I see, so I would download this file now and include it as part of the rock as opposed to download it from the archive when building each time.

Reply:
@gtato if we do the `dump` plugin using `source` per your other suggestion I don't know that it is still possible to compare to the `sha512` since the tar is automatically unpacked

Reply:
Using `source-checksum` has resolved this.